app/api/files/record/route.ts

```diff
@@ -0,0 +1,48 @@
+import { NextResponse } from "next/server";
+import supabase from "@/lib/supabase/serverClient";
+
+export async function POST(req: Request) {
+  try {
+    const body = await req.json();
+    const {
+      ownerAccountId,
+      artistAccountId,
+      storageKey,
+      fileName,
+      mimeType,
+      sizeBytes,
+      description,
+      tags,
+    } = body || {};
+
+    if (!ownerAccountId || !artistAccountId || !storageKey || !fileName) {
+      return NextResponse.json({ error: "Missing required fields" }, { status: 400 });
+    }
+
+    const { data, error } = await supabase
+      .from("files")
+      .insert({
+        owner_account_id: ownerAccountId,
+        artist_account_id: artistAccountId,
+        storage_key: storageKey,
+        file_name: fileName,
+        mime_type: mimeType ?? null,
+        size_bytes: sizeBytes ?? null,
+        description: description ?? null,
+        tags: Array.isArray(tags) ? tags : [],
+      })
+      .select("*")
+      .single();
```

Comment:
Single responsibility principle
- actual: defining supabase queries in an API endpoint
- required: standalone files for supabase queries